media-libs/openexr/openexr-2.5.2.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_ECLASS=cmake
+inherit cmake-multilib flag-o-matic toolchain-funcs
+
+DESCRIPTION="ILM's OpenEXR high dynamic-range image file format libraries"
+HOMEPAGE="https://www.openexr.com/"
+SRC_URI="https://github.com/AcademySoftwareFoundation/openexr/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/25" # based on SONAME
+KEYWORDS="~amd64 -arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~x64-macos ~x86-solaris"
+IUSE="cpu_flags_x86_avx doc examples static-libs utils test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	media-libs/ilmbase:=
+	sys-libs/zlib[${MULTILIB_USEDEP}]
+"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+S="${WORKDIR}/${P}/OpenEXR"
+
+DOCS=( PATENTS README.md )
+
+MULTILIB_WRAPPED_HEADERS=( /usr/include/OpenEXR/OpenEXRConfigInternal.h )
+
+src_prepare() {
+	# Fix path for testsuite
+	sed -i -e "s:/var/tmp/:${T}:" "${S}"/IlmImfTest/tmpDir.h || die "failed to set temp path for tests"
+
+	if use abi_x86_32 && use test; then
+		eapply "${FILESDIR}/${P}-0001-IlmImfTest-main.cpp-disable-tests.patch"
+	fi
+
+	cmake_src_prepare
```

Comment:
In general, I recommend running `cmake_src_prepare` before anything else in src_prepare so that patches are applied over vanilla sources.

Reply:
Hmm, this make the round-trip of applying patches, test and eventually change them more complicated and introduces extra steps necessary when preparing patches for upstream.

Reply:
Not sure I can follow that. All modification of source done in ebuild instead of patch should go below cmake_src_prepare.

You prepare upstreaming patches either in a git clone of upstream source, or on vanilla sources (after src_unpack, not src_prepare) anyway.

Reply:
> I assume non-patch changes are not ones you try to upstream.

Why not? Sometimes it's faster to just 'patch' using a few `sed` statements, which could, nonetheless, be useful by upstream and be packged together in a patch.

> You prepare upstreaming patches either in a git clone of upstream source, or on vanilla sources (after src_unpack, not src_prepare) anyway.

A clone, might not always be the way to go, if there have been several updates already to upstream's sources. I usually unpack and make these vanilla sources a local git repo before I do anything else. It's not much of an issue, it might just involve an additional commit and/or commenting patches already present in the ebuild in some circumstances

Reply:
In general, sed is discouraged because it will fail silently. We still use it because it is hard to quit bad habits and yes, sometimes it is so damn comfortable. ;) If a patch is pending upstream though, I will use that instead, and expect a reference to that or a note that it is in some future version - then the same should be done for the `sed`.

> I usually unpack and make these vanilla sources a local git repo before I do anything else.

In that case you are already screwed after src_prepare because you don't have vanilla sources anymore. Vanilla they are only after src_unpack, then you can apply whatever patches you have in mind after initialising a git repo. I've done that as well previously, but personally for packages I am registered as maintainer I keep a git clone - nothing easier than branching off a certain git tag for our particular version use case of even maintaining a 'gentoo' branch for non-upstreamable patches for easy rebasing.

Reply:
> I've done that as well previously, but personally for packages I am registered as maintainer I keep a git clone - nothing easier than branching off a certain git tag for our particular version use case or even maintaining a 'gentoo' branch for non-upstreamable patches for easy rebasing.

Sounds like a good approach. I will think about it and give it a try to establish a reasonable workflow.